spark/src/main/scala/org/apache/spark/sql/delta/DeltaHistoryManager.scala

```diff
@@ -115,8 +119,61 @@ class DeltaHistoryManager(
           }
         }.map(DeltaHistory.fromCommitInfo)
       }
-    // Spark should return the commits in increasing order as well
-    monotonizeCommitTimestamps(info.collect()).reverse
+    val monotonizedCommits = if (useInCommitTimestamps) {
+      // ICT timestamps are guaranteed to be monotonically increasing.
+      info.collect()
+    } else {
+      monotonizeCommitTimestamps(info.collect())
+    }
+    monotonizedCommits.reverse
+  }
+
+  /**
+   * Get the commit information of the Delta table from commit `[start, end]` in reverse
+   * chronological order. If `end` is `None`, we return all commits from start to now.
+   * @param start The start of the commit range, inclusive.
+   * @param end The end of the commit range, inclusive.
+   */
+  def getHistory(
+      start: Long,
+      end: Option[Long] = None): Seq[DeltaHistory] = {
+    val currentSnapshot = deltaLog.unsafeVolatileSnapshot
+    val (snapshot, resolvedEnd) = end match {
```

Comment:
`snapshot` -> `snapshotForCommitFileProvider` since the snapshot might not necessarily be useful for other usecases.

Reply:
Renamed to `snapshotNewerThanResolvedEnd`.